src/main/java/com/iexec/worker/tee/TeeServicesPropertiesService.java

```diff
@@ -74,7 +75,12 @@ <T extends TeeServicesProperties> T retrieveTeeServicesProperties(final String c
                             ", actualFramework:" + smsTeeFramework + "]");
         }
 
-        final T properties = smsClient.getTeeServicesProperties(teeFramework);
+        final TeeEnclaveConfiguration teeEnclaveConfiguration = taskDescription.getAppEnclaveConfiguration();
+        if (teeEnclaveConfiguration == null) {
```

Comment:
Alternatively, we may avoid the `if` here and use this:
```suggestion
        Objects.requireNonNull(teeEnclaveConfiguration, "Missing TEE enclave configuration [chainTaskId:" + chainTaskId + "]");
```
See https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Objects.html#requireNonNull(T,java.lang.String)